### ISSUE TITLE
Dashboard Exception "..does not implement IController." Sitecore 9.0.1

### ISSUE THREAD

**Issue by WizX20** [bmbruno/SitecoreSpark.SitecorePublishReportingKit#1]
After clean installation, also moved the configs to _App_Config\Environment_ to make sure that they are loaded last.

_Sitecore 9.0.1 rev. 171219 (WDP XP0 packages)
Sitecore PowerShell extensions 4.7.2
Sitecore Experience Accelerator 1.6 rev. 180103_

```
An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.
--
[ExceptionWrapper: The controller for path '/sitecore/sprk' was not found or does not implement IController.]

[ControllerCreationException: Could not create controller: 'Report'. 
The current route url is: 'sitecore/sprk/{controller}/{action}/{id}'. ]
   Sitecore.Mvc.Controllers.SitecoreControllerFactory.CreateController(RequestContext requestContext, String controllerName) +171
   System.Web.Mvc.MvcHandler.ProcessRequestInit(HttpContextBase httpContext, IController& controller, IControllerFactory& factory) +265
   System.Web.Mvc.MvcHandler.BeginProcessRequest(HttpContextBase httpContext, AsyncCallback callback, Object state) +77
   System.Web.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +909
   System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step) +195
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +128
```

**Comment by bmbruno** [comment]
Thanks for the clarification. I'll reproduce this issue and make the needed corrections. I'll be sure to test it with a full SxA install.

**Comment by bmbruno** [comment]
I'm unable to reproduce this exact error with the setup you've provided.

One possible issue: the SPRK routing file does not get patched where I intended it to be patched. Instead, `/App_Config/Include/Z.LayoutService.LayoutService.Mvc.config` gets applied just before the SPRK patch file. This doesn't appear to be causing an issue.

Is this a purely clean installation of Sitecore, as you say, or is there any other configuration for custom routes?

**Comment by bmbruno** [comment]
This may actually be a bug related to controller injection in Sitecore 9.0.1, so additional testing will be needed. I'll try to address this with the upcoming 1.5 release this fall.

**Comment by bmbruno** [comment]
This will be fixed with the upcoming `1.5.0` release. There has been a change in DI registration for the ReportController.
